all_test.go

```diff
@@ -351,6 +362,9 @@ func TestOptions_PreserveOwner(t *testing.T) {
 }
 
 func TestOptions_CopyRateLimit(t *testing.T) {
+	if defaultCopyMethodName != "CopyBytes" {
+		t.Skipf("Can only test WrapReader with CopyBytes method, not %s", defaultCopyMethodName)
+	}
```

Comment:
This doesn't seem the exception of test cases, but it is an expected behavior. In other words, this package should return error if users try to use `CopyRateLimit` when invalid copy-method is set.

Reply:
> This package should return error if users try to use CopyRateLimit when invalid copy-method is set.

Yes.

I should clarify: With this pull request, I'm re-uing tests made for `CopyBytes` to check that from the user perspective, other `FileCopyMethod`s behave like `CopyBytes` where possible and do not have incorrect behaviour.

This and the `TestOptions_FS` tests are skipped because the package does return an error, which is expected behaviour. They need to be handled specially or else `go test` would always fail.

Instead of skipping, do you want me to check that they correctly return the error?

Reply:
First of all, please do not say sorry, nor apologize. You're doing great contribution and I really appreciate your effort and passion. Being honest and frank is very important element to keep OSS clean.

> correctly return the error

Well, yes, if it should do so, we should do so.